### ISSUE TITLE
Investigate if crossdac packaging can be built per configuration

### ISSUE THREAD

**Issue by steveisok** [dotnet/runtime#76912]
In order to make the runtime more source build friendly, we need to try to eliminate as many "join points" as possible in the official build.  We need to determine if it's even possible to split out crossdac packaging per configuration.

**Comment by jkoritzinsky** [comment]
With the new support for step targets and the recent refactoring I did for container resources, as well as Linux Containers on Windows support, we have the ability to make the cross-dac packaging step fully independent of the build job, or alternatively, do the cross-dac build as part of the product build.

**Comment by steveisok** [comment]
Thanks @jkoritzinsky!  Good enough for me.

**Comment by jkoritzinsky** [comment]
Do we want to keep this open to track doing this work, or do we want to open a new issue tracking the work?

**Comment by ViktorHofer** [comment]
Let's keep this open to track the work.

**Comment by ViktorHofer** [comment]
@jkoritzinsky when you say step targets, are you referring to [`templateContext`](https://learn.microsoft.com/en-us/azure/devops/release-notes/2022/sprint-202-update#extended-yaml-pipelines-templates-can-now-be-passed-context-information-for-stages-jobs-and-deployments)?

**Comment by jkoritzinsky** [comment]
No, I’m referring to the ability to use the `target` property on a step to run it in a different named container. Here’s the place we use it today:

https://github.com/dotnet/runtime/blob/6ee57c0d1a8ef6ea6a5886b3367f04702fa07e58/eng/pipelines/installer/jobs/steps/build-linux-package.yml#L22

**Comment by jkoritzinsky** [comment]
cc: @hoyosjs 

In the infra standup today, we discussed moving the cross-DAC build/package job into a separate pipeline along with the workloads job as both of them produce intermediate artifacts that we do not ship to customers (cross-DAC is for the symbol server only and the workloads job only produces assets that we insert into Visual Studio).
